File: csharp/RIoT.Sdk.Core/OrderRef.cs
namespace RIoT.Sdk.Core;

/// <summary>
/// Identifiers for an order created via byDefaultMissions (BC-ORDER-001 / BC-ORDER-005).
/// Id = numeric PK; OrderId = string order-...; UpperId = caller correlation id.
/// </summary>
public sealed record OrderRef(int Id, string OrderId, string UpperId, int OrderState);

File: python/riot_sdk/core/order_ref.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class OrderRef:
    """Identifiers for an order from byDefaultMissions (BC-ORDER-001 / BC-ORDER-005)."""

    id: int
    order_id: str
    upper_id: str
    order_state: int
